### PR TITLE
update CI to rust 1.56

### PR DESCRIPTION
**Pull request by michaelkirk**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- [n/a] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Corresponding PR at https://github.com/georust/docker-images/pull/14

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -34,7 +34,14 @@ jobs:
         working-directory: geo-types
     strategy:
       matrix:
-        container_image: ["georust/geo-ci:rust-1.51", "georust/geo-ci:rust-1.52", "georust/geo-ci:rust-1.53"]
+        container_image:
+          # We aim to support at least the last 6 months of rust toolchains, so
+          # run on MSRV and the most recent two releases.
+          #
+          # We omit some of the middle releases for faster CI runs.
+          - "georust/geo-ci:rust-1.51"
```
Comment (michaelkirk) [inline .github/workflows/test.yml:42]: If we had an MSRV of 1.52, it'd still be within a 6 month threshold, but it seemed like we might as well maintain support for 1.51 if tests are already passing.

Also, I realize this "6 month support" policy could be controversial. In practice we've been doing something pretty close to this anyway, but I'd be open to revisiting it if it proves arduous.

File: geo/fuzz/Cargo.toml [hunk 2]
```diff
@@ -12,9 +12,11 @@ cargo-fuzz = true
 libfuzzer-sys = "0.4"
 
 [dependencies.geo]
+path = ".."
```
Comment (michaelkirk) [inline geo/fuzz/Cargo.toml:15]: specifying a dependency without a version or path was causing a warning (which became an error in rust-1.55).

File: .github/workflows/test.yml (hunk 1 above)
Comment (urschrei) [reply .github/workflows/test.yml:42]: Very happy to throw it open for wider discussion, but the 6 month policy seems reasonable to me (as someone who admittedly doesn't depend on my distro's Rust version)

**Review by lnicola — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (lnicola) [reply .github/workflows/test.yml:42]: I would slightly prefer something more precise than "6 months", like "stable - 3".

Comment (michaelkirk) [reply .github/workflows/test.yml:42]: > I would slightly prefer something more precise than "6 months", like "stable - 3".

Good idea - that would be easier to think about. Updated!

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -71,7 +87,15 @@ jobs:
         working-directory: geo-postgis
     strategy:
       matrix:
-        container_image: ["georust/geo-ci:rust-1.51", "georust/geo-ci:rust-1.52", "georust/geo-ci:rust-1.53"]
+        container_image:
+          # We aim to support rust-stable plus (at least) the prior 3 releases,
+          # giving us about 6 months of coverage.
+          #
+          # Minimum supported rust versino (MSRV)
```
Comment (lnicola) [inline .github/workflows/test.yml:94]:
```suggestion
          # Minimum supported rust version (MSRV)
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -52,7 +60,15 @@ jobs:
         working-directory: geo
     strategy:
       matrix:
-        container_image: ["georust/geo-ci:rust-1.51", "georust/geo-ci:rust-1.52", "georust/geo-ci:rust-1.53"]
+        container_image:
+          # We aim to support rust-stable plus (at least) the prior 3 releases,
+          # giving us about 6 months of coverage.
+          #
+          # Minimum supported rust versino (MSRV)
```
Comment (lnicola) [inline .github/workflows/test.yml:67]:
```suggestion
          # Minimum supported rust version (MSRV)
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -34,7 +34,15 @@ jobs:
         working-directory: geo-types
     strategy:
       matrix:
-        container_image: ["georust/geo-ci:rust-1.51", "georust/geo-ci:rust-1.52", "georust/geo-ci:rust-1.53"]
+        container_image:
+          # We aim to support rust-stable plus (at least) the prior 3 releases,
+          # giving us about 6 months of coverage.
+          #
+          # Minimum supported rust versino (MSRV)
```
Comment (lnicola) [inline .github/workflows/test.yml:41]:
```suggestion
          # Minimum supported rust version (MSRV)
```

**Comment by michaelkirk** [comment]
bors r=lnicola

**Pull request merged**